src/cuda/api/memory.hpp

```diff
@@ -494,7 +494,7 @@ inline void copy(void *destination, const void *source, size_t num_bytes, stream
  * @param source A pointer to a a memory region of size `destination.size() * sizeof(T)`
  */
 template <typename T, size_t NDIMS, bool StreamIsOnCurrentDevice>
-inline void copy(array::array_t<T, NDIMS>& destination, const void *source, stream_t<StreamIsOnCurrentDevice>& stream);
+inline void copy(const array::array_t<T, NDIMS>& destination, const void *source, stream_t<StreamIsOnCurrentDevice>& stream);
```

Comment:
It is possible because a `cudaArray_t` seems to be a view on a memory location. Thus you can do sth like this:

```
const cudaArray_t carr;
cudaArray_t arr;
...
arr = carr;
// arr and carr point to the same array
```

But we can omit the `const` on the destination array for better readability.

Reply:
Yes, you're right. I think I've probably done the same thing in a few places with streams or devices (i.e. their data fields don't change but the stream state changes).